extensions/amp-story/0.1/media-pool.js

```diff
@@ -205,21 +205,28 @@ export class MediaPool {
     this.forEachMediaType_(key => {
       const type = MediaType[key];
       const count = maxCounts[type] || 0;
+
+      // Small optimization. Cloning nodes is faster than building them, so
+      // constructing a seed element makes sense for initialization.
+      const seed = this.mediaFactory_[type].call(this);
+
       this.allocated[type] = [];
       this.unallocated[type] = [];
-      for (let i = 0; i < count; i++) {
-        this.vsync_.mutate(() => {
-          const mediaEl = this.mediaFactory_[type].call(this);
-          mediaEl.setAttribute('pool-element', elId++);
+
+      this.vsync_.mutate(() => {
+        for (let i = count; i > 0; i--) {
```

Comment:
By default, Closure would compile into this optimized reverse loop. However, since a comparison against `i` is done below, the compiler can no longer make that assumption.